src/components/WalletModal/index.tsx

```diff
@@ -185,9 +126,9 @@ export default function WalletModal({ openSettings }: { openSettings: () => void
         <ThemedText.SubHeader fontWeight={500}>Connect a wallet</ThemedText.SubHeader>
         <IconButton Icon={Settings} onClick={openSettings} data-testid="wallet-settings" />
       </AutoRow>
-      {pendingError ? (
-        pendingConnection && (
-          <ConnectionErrorView openOptions={openOptions} retryActivation={() => tryActivation(pendingConnection)} />
+      {error ? (
+        pendingConnection !== undefined && (
```

Comment:
If we manage to clear `error` automatically when `isPending`, we don't need this nested `if` condition, right? We can assume that errors will get automatically cleared when activation is called. This is what often happens with forms, where `error` is cleared as soon as you start typing, so you don't check for text input length (I hope that analogy works here)

Reply:
I refactored the return type to make this cleaner 👍